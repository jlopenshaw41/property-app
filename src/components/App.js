import React from "react";
import "../styles/App.css";
import { Route, Switch } from "react-router-dom";
import NavBar from "./NavBar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";

const App = () => {
  return (
    <div className="App">
      <NavBar />
      <Switch>
        <Route exact path="/">
          <Properties />
        </Route>
        <Route exact path="/add-property">
          <AddProperty />
        </Route>
      </Switch>
      <h2>Property App</h2>
    </div>
  );
};

export default App;
