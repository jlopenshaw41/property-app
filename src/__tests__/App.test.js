import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  it("renders as expected", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders property app text", () => {
    render(<App />);
    const appName = screen.getByText("Property App");
    expect(appName).toBeInTheDocument();
  });
});
