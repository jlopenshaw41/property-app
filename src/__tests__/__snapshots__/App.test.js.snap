// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navbar"
    >
      <img
        alt="Logo"
        data-testid="logo"
        src="logo.png"
      />
      <div
        class="navbar-links"
      >
        <ul>
          <li
            class="navbar-links-item"
            data-testid="nav-link-1"
          >
            <a
              class="link"
              href="/"
            >
              View properties
            </a>
          </li>
          <li
            class="navbar-links-item"
            data-testid="nav-link-2"
          >
            <a
              class="link"
              href="/add-property"
            >
              Add a property
            </a>
          </li>
        </ul>
      </div>
    </div>
    <div>
      Properties page
      <div
        class="property-card"
      >
        <div
          class="property-card-title"
        >
          Example property text
        </div>
        <div
          class="property-card-type"
        >
          Cottage
        </div>
        <div
          class="property-card-bedrooms"
          data-testid="property-card-bedrooms"
        >
          3
        </div>
        <div
          class="property-card-bathrooms"
          data-testid="property-card-bathrooms"
        >
          2
        </div>
        <div
          class="property-card-price"
          data-testid="property-card-price"
        >
          300000
        </div>
        <div
          class="property-card-city"
        >
          Manchester
        </div>
        <div
          class="property-card-email"
        >
          <a
            href="mailto:email@address.com"
          >
            email@address.com
          </a>
        </div>
      </div>
    </div>
    <h2>
      Property App
    </h2>
  </div>
</DocumentFragment>
`;
